### PR TITLE
F1 Menu: Moved Further ConVars

### PR DESCRIPTION
**Pull request by TimGoll**
### Added

- Added all TTT2 convars into the F1 menu
  - most convars are located in the 'administration' menu
  - equipment specific settings can be found in the 'edit equipment' menu
- Added icon to the magneto stick
- Added the function `AddToSettingsMenu` to both `SWEP` and `ITEM` to add settings to the equipment menu
- Added some TTT entities to the TTT2 base to modify their code
- Added missing docs where spotted & small code style cleanup

### Changed

- Changed logs folder to `terrortown/logs/` to be inline with everything else
- Added more role agnostics
  - voice drain rate is now no longer bound to Detectives but to all public policing roles
  - Karma multiplier is now no longer bound to Detectives but to all public policing roles
  - all non-innocent roles are now able to pin ragdolls if enabled (previous only Traitors could do this)
- Added convar flags to convars that were missing them
- Removed legacy mstack code for the teleporter now that its code is up to date

### Breaking Changes

- Renamed some convars to be inline with our 'opt-in style', all values were changed so that the default value is kept
  - `ttt_no_prop_throwing` is now `ttt_prop_throwing`
  - `ttt_limit_spectator_chat` is now `ttt_spectators_chat_globally`
  - `ttt_no_nade_throw_during_prep` is now `ttt_nade_throw_during_prep`
  - `ttt_armor_classic` is now `ttt_armor_dynamic`

ConVars that will be added at a later point:

- `ttt_playercolor_mode (0..4)` [player models]
- `ttt_ban_type` (it was never in ULX, maybe it is not that important)
- maybe some other variables that were missing in ulx, I only knew about a few of them

### CODE REVIEW

File: lua/terrortown/menus/gamemode/equipment/base_equipment.lua [hunk 1]
```diff
@@ -99,4 +99,14 @@ function CLGAMEMODESUBMENU:Populate(parent)
 
 		masterRefs[key] = option
 	end
+
+	-- now add custom equipment settings
+	if not isfunction(equipment.AddToSettingsMenu) then
```
Comment (saibotk) [inline lua/terrortown/menus/gamemode/equipment/base_equipment.lua:104]: isn't this true ("false" here) then for all weapons / items (that use the ttt base), because the function is defined on the base as an empty function?

File: lua/terrortown/menus/gamemode/equipment/base_equipment.lua [hunk 2]
```diff
@@ -99,4 +99,14 @@ function CLGAMEMODESUBMENU:Populate(parent)
...
+		print(equipment.PrintName)
+
+		PrintTable(equipment)
```
Comment (saibotk) [inline lua/terrortown/menus/gamemode/equipment/base_equipment.lua:107]: Maybe some info about these prints? 

Eg. print something like "Ohh not adding equip to settings cause: ...." 
Maybe even omit it?

File: gamemodes/terrortown/gamemode/client/cl_shopeditor.lua [hunk 3]
```diff
@@ -63,6 +63,9 @@ function ShopEditor.GetEquipmentForRoleAll()
 		local eq = weps[i]
 		local name = WEPS.GetClass(eq)
 
+		-- Get Inheritable version
+		eq = weapons.Get(name)
+
```
Comment (saibotk) [inline gamemodes/terrortown/gamemode/client/cl_shopeditor.lua:68]: Why is this included in this PR? 

Should be explained or removed from this one

**Review by saibotk — CHANGES_REQUESTED** [review]

File: lua/terrortown/menus/gamemode/equipment/base_equipment.lua (hunk 1 above)
Comment (TimGoll) [reply lua/terrortown/menus/gamemode/equipment/base_equipment.lua:104]: This PR is WIP and this is just for debugging

File: gamemodes/terrortown/gamemode/client/cl_shopeditor.lua (hunk 3 above)
Comment (TimGoll) [reply gamemodes/terrortown/gamemode/client/cl_shopeditor.lua:68]: This is to get the complete class. With inheritance from the base

File: lua/terrortown/lang/en.lua [hunk 4]
```diff
@@ -1813,3 +1814,8 @@ L.label_max_unarmed_slots = "Max unarmed slots"
 L.label_max_special_slots = "Max special slots"
 L.label_max_extra_slots = "Max extra slots"
 L.label_weapon_autopickup = "Enable automatic weapon pickup"
+L.label_sprint_enabled = "Enable sprinting"
+L.label_sprint_max = "Max sprinting stamina"
+L.label_sprint_stamina_consumption = "Stamina consumtion factor"
```
Comment (Histalek) [inline lua/terrortown/lang/en.lua:1819]:
```suggestion
L.label_sprint_stamina_consumption = "Stamina consumption factor"
```

File: lua/terrortown/lang/en.lua [hunk 5]
```diff
@@ -108,7 +108,7 @@ L.disg_menutitle = "Disguise control"
 L.disg_not_owned = "You are not carrying a Disguiser!"
 L.disg_enable = "Enable disguise"
 
-L.disg_help1 = "When your disguise is active, your name, health and karma do not show when someone looks at you. In addition, you will be hidden from a Detective's radar."
+L.disg_help1 = "When your disguise is active, your name, health and Karma do not show when someone looks at you. In addition, you will be hidden from a Detective's radar."
```
Comment (Histalek) [inline lua/terrortown/lang/en.lua:111]: What is the reason for capitalizing this?

Comment (TimGoll) [reply lua/terrortown/lang/en.lua:111]: Because on the official TTT website Karma is always upper case. Therefore I replaced it in the whole translation

**Review by Histalek — APPROVED** [review]

File: gamemodes/terrortown/entities/weapons/weapon_zm_improvised.lua [hunk 6]
```diff
@@ -10,11 +10,11 @@ if SERVER then
 
 	---
 	-- @realm server
-	cv_crowbar_unlocks = CreateConVar("ttt_crowbar_unlocks", "1", FCVAR_ARCHIVE)
+	cv_crowbar_unlocks = CreateConVar("ttt_crowbar_unlocks", "1", {FCVAR_ARCHIVE, FCVAR_NOTIFY})
 
 	---
 	-- @realm server
-	cv_crowbar_pushforce = CreateConVar("ttt_crowbar_pushforce", "395", FCVAR_NOTIFY)
+	cv_crowbar_pushforce = CreateConVar("ttt_crowbar_pushforce", "395", {FCVAR_ARCHIVE, FCVAR_NOTIFY})
```
Comment (saibotk) [inline gamemodes/terrortown/entities/weapons/weapon_zm_improvised.lua:17]: What about including those two?

File: gamemodes/terrortown/gamemode/client/cl_lang.lua [hunk 7]
```diff
@@ -535,23 +535,7 @@ end
...
-	},
-
-	chat_warn = { -- TODO move to teleporter
-		"tele_no_ground",
-		"tele_no_crouch",
-		"tele_no_mark_ground",
-		"tele_no_mark_crouch"
-	}
-}
+-- This is kept for compatibility reasons so that old addons may still use this.
```
Comment (saibotk) [inline gamemodes/terrortown/gamemode/client/cl_lang.lua:538]: what exactly?

File: lua/terrortown/entities/items/item_ttt_armor.lua [hunk 8]
```diff
@@ -36,4 +36,75 @@ else -- CLIENT
...
+		})
+
+		form:MakeHelp({
+			label = "help_item_armor_classic"
+		})
+
+		form:MakeCheckBox({
+			serverConvar = "ttt_armor_classic",
+			label = "label_armor_classic"
+		})
```
Comment (saibotk) [inline lua/terrortown/entities/items/item_ttt_armor.lua:66]: I think we should only put settings in here that are directly related to the specific item and not the armor system in general.

File: lua/terrortown/lang/de.lua [hunk 9]
```diff
@@ -438,8 +438,8 @@ L.tele_no_mark = "Keine Stelle markiert. Markiere ein Ziel vor dem Teleportieren
 L.tele_no_mark_ground = "Kann keine Teleportstelle markieren, während man nicht auf solidem Untergrund steht!"
 L.tele_no_mark_crouch = "Kann keine Teleportstelle markieren, während man geduckt ist!"
 
-L.tele_help_pri = "{primaryfire} teleportiert dich zur markierten Stelle."
-L.tele_help_sec = "{secondaryfire} markiert momentane Position."
+L.tele_help_pri = "Teleportiert dich zur markierten Stelle"
+L.tele_help_sec = "{Markiert momentane Position"
```
Comment (saibotk) [inline lua/terrortown/lang/de.lua:442]:
```suggestion
L.tele_help_sec = "Markiert momentane Position"
```

**Review by saibotk — CHANGES_REQUESTED** [review]

File: gamemodes/terrortown/gamemode/client/cl_lang.lua (hunk 7 above)
Comment (TimGoll) [reply gamemodes/terrortown/gamemode/client/cl_lang.lua:538]: image
Read the whole block

File: gamemodes/terrortown/entities/weapons/weapon_zm_improvised.lua (hunk 6 above)
Comment (TimGoll) [reply gamemodes/terrortown/entities/weapons/weapon_zm_improvised.lua:17]: I don't understand what you mean

File: lua/terrortown/entities/items/item_ttt_armor.lua (hunk 8 above)
Comment (TimGoll) [reply lua/terrortown/entities/items/item_ttt_armor.lua:66]: I mean, the balancing settings are related to the item, aren't they?

File: gamemodes/terrortown/entities/weapons/weapon_zm_improvised.lua (hunk 6 above)
Comment (saibotk) [reply gamemodes/terrortown/entities/weapons/weapon_zm_improvised.lua:17]: They aren't included in the menu are they?

Comment (TimGoll) [reply gamemodes/terrortown/entities/weapons/weapon_zm_improvised.lua:17]: you are absolutely right, I will add them as well

File: lua/terrortown/entities/items/item_ttt_armor.lua (hunk 8 above)
Comment (saibotk) [reply lua/terrortown/entities/items/item_ttt_armor.lua:66]: But not only to the item and it's rather unintuitive to look for settings like armor at begin or general classic/new armor system in the armor item. Other items might also use the armor system

Comment (TimGoll) [reply lua/terrortown/entities/items/item_ttt_armor.lua:66]: Armor at begin isn't located in this menu for this very reason. If other items also use the armor system, what does it change?

Comment (saibotk) [reply lua/terrortown/entities/items/item_ttt_armor.lua:66]: If this is not included then why include settings that are related to the armor system in general on a specific item instead of including it in gameplay settings where a user expects them?

Comment (TimGoll) [reply lua/terrortown/entities/items/item_ttt_armor.lua:66]: Because "armor on spawn" is a gameplay setting. The handling of the armor itself is a item setting. At least in my opinion the user expects armor settings within the armor item. Especially since crowbar, DNA scanner, teleporter, ... settings are all located in their items instead of the gameplay menu. Why should the armor be different?

Comment (saibotk) [reply lua/terrortown/entities/items/item_ttt_armor.lua:66]: Because armor is a much more general system than the dna scanner etc, thats just my opinion. 
Any one else want to comment on this? @Histalek or @ZenBre4ker ?

Comment (ZenBre4ker) [reply lua/terrortown/entities/items/item_ttt_armor.lua:66]: I dont know much about armor, but I agree that placing it in the item-system is really unintuitive, Id prefer general armor settings in a gameplay menu. And only stuff that classically relates to a buyable item should be here. Its weird, that armor is an item anyway. For me that would be two different things. An item I can buy to get armor in the first place and a general system that decides who gets the armor at the start of a round depending on roles.

File: lua/terrortown/lang/en.lua [hunk 10]
```diff
@@ -1622,3 +1622,200 @@ L.label_roles_credits_award_kill = "Credit reward amount for kill"
...
+If a player changes their name during a round, this can be abused to evade being killed. Therefore it is prohibited to change the nickname during an active round.
+
+If the bantime is greater than 0, the player will be unable to reconnect to the server until that time has passed.]]
+L.help_damage_log = "Each time a player is damaged, a damage log entry is added to the console if enabled. This can also be stored to disk after a round has ended. The file is located at 'data/terrortown/logs/'"
+L.help_spawn_waves = [[
+If this variable is set to 0, all players are spawned at once. For servers with huge amounts of players, it can be beneficial to spawn the players in waves. The spawn wave interval is the time between each spawn wave. A spawn wave always spawns as many players as there are valid spawn points.
+
+Note: Make sure that the preparing time is long enough for the desired amount of spawn waves.]]
+L.help_voicechat_battery = [[
+Voicechatting with enabled voice chat battery reduces this meter. When it's empty the player can't voicechat and must wait for a few seconds for it to recharge. This can help to prevent excessive voicechat usage.
```
Comment (ZenBre4ker) [inline lua/terrortown/lang/en.lua:1676]: Its the main usage for "müssen", must is more a should do it. In this case he really has to wait for it.
```suggestion
Voicechatting with enabled voice chat battery reduces this meter. When it's empty the player can't voicechat and has to wait for a few seconds for it to recharge. This can help to prevent excessive voicechat usage.
```

File: lua/terrortown/lang/en.lua [hunk 11]
```diff
@@ -1622,3 +1622,200 @@ L.label_roles_credits_award_kill = "Credit reward amount for kill"
...
+
+When taking damage, a certain percentage of this damage is converted into armor damage, a different percentage is still applied to the player and the rest vanishes.
+
+If reinforced armor is enabled, the damage applied to the player is decreased by 15% as long as the armor value is above the reinforcement threshold.]]
+L.help_sherlock_mode = "The sherlock mode is the classic TTT mode. If the sherlock mode is disabled, dead bodies can not be confirmed, the scoreboard shows everyone as alive and the spectators can talk to the living players."
+L.help_prop_possession = [[
+Prop possession can be used by spectators to possess props lying in the world and use the slowly recharging 'punch-o-meter' to move said prop around.
+
+The maximum value of the 'punch-o-meter' consists of a possession base value, where the kills/deaths difference clamped inbetween two defined limmits is added. The meter slowly recharges over time. The set recharge time is the time needed to recharge a single point in the 'punch-o-meter'.]]
```
Comment (ZenBre4ker) [inline lua/terrortown/lang/en.lua:1698]:
```suggestion
The maximum value of the 'punch-o-meter' consists of a possession base value, where the kills/deaths difference clamped inbetween two defined limits is added. The meter slowly recharges over time. The set recharge time is the time needed to recharge a single point in the 'punch-o-meter'.]]
```

File: lua/terrortown/lang/en.lua [hunk 12]
```diff
@@ -1622,3 +1622,200 @@ L.label_roles_credits_award_kill = "Credit reward amount for kill"
...
+L.help_karma_bonus = "There are also two different passive ways to gain Karma during a round. First a round heal is applied to every player. Then a secondary clean bonus is given if no teammates were hurt or killed."
+L.help_karma_clean_half = [[
+When a player's Karma is above the starting level (meaning the Karma max has been configured to be higher than that), all their Karma increases will be reduced based on how far their Karma is above that starting level. So it goes up slower the higher it is.
+
+This reduction goes in a curve of exponential decay: initially it's fast, and it slows down as the increment gets smaller. This convar sets at what point the bonus has been halved (so the half-life). With the default value of 0.25, if a the starting amount of Karma is 1000 and the max 1500, and a player has Karma 1125 ((1500 - 1000) * 0.25 = 125), then his clean round bonus will be 30 / 2 = 15. So to make the bonus go down faster you’d set this convar lower, to make it go down slower you’d increase it towards 1.]]
```
Comment (ZenBre4ker) [inline lua/terrortown/lang/en.lua:1708]:
```suggestion
This reduction goes in a curve of exponential decay: initially it's fast, and it slows down as the increment gets smaller. This convar sets at what point the bonus has been halved (so the half-life). With the default value of 0.25, if the starting amount of Karma is 1000 and the max 1500, and a player has Karma 1125 ((1500 - 1000) * 0.25 = 125), then his clean round bonus will be 30 / 2 = 15. So to make the bonus go down faster you’d set this convar lower, to make it go down slower you’d increase it towards 1.]]
```

File: lua/terrortown/lang/en.lua [hunk 13]
```diff
@@ -1622,3 +1622,200 @@ L.label_roles_credits_award_kill = "Credit reward amount for kill"
...
+
+L.label_killer_dna_range = "Max kill range to leave DNA"
+L.label_killer_dna_basetime = "Sample life base time"
+L.label_dna_scanner_slots = "DNA sample slots"
+L.label_dna_radar = "Enable classic DNA scanner mode"
+L.label_dna_radar_cooldown = "DNA scanner cooldown"
+L.label_radar_charge_time = "Recharge time after a radar sample"
+L.label_crowbar_shove_delay = "Cooldown after crowbar push"
+L.label_idle = "Enable idle mode"
+L.label_idle_limit = "Maximal idle time in seconds"
```
Comment (ZenBre4ker) [inline lua/terrortown/lang/en.lua:1719]: As this time is the longest possible idle time its 'maximum' instead of 'maximal'. Had to google it, because maximal sounded wrong. 😄 
```suggestion
L.label_idle_limit = "Maximum idle time in seconds"
```

File: lua/terrortown/menus/gamemode/equipment.lua [hunk 14]
```diff
@@ -33,9 +33,15 @@ function CLGAMEMODEMENU:InitializeVirtualMenus()
 
 		counter = counter + 1
 
+		local isItem = items.IsItem(equipment)
+
+		if not isItem then
+			equipment = weapons.Get(WEPS.GetClass(equipment))
+		end
```
Comment (ZenBre4ker) [inline lua/terrortown/menus/gamemode/equipment.lua:40]: I forgot a comment here:
```suggestion
		-- Get inheritable version for weapons
		if not isItem then
			equipment = weapons.Get(WEPS.GetClass(equipment))
		end
```

File: lua/terrortown/menus/gamemode/administration/hud.lua [hunk 15]
```diff
@@ -71,6 +72,20 @@ function CLGAMEMODESUBMENU:Populate(parent)
 			end
 		})
 	end
+
+	local form3 = vgui.CreateTTT2Form(parent, "header_hud_toggleable")
+
+	for i = 1, #hudElemList do
+		local elem = hudElemList[i]
+
+		if not elem.togglable then continue end
```
Comment (ZenBre4ker) [inline lua/terrortown/menus/gamemode/administration/hud.lua:81]: is that correct? not toggl e able? Both is possible, just noticed. Dont know which one is used in code

**Review by ZenBre4ker — APPROVED** [review]
Great work as always <3 Beside a few misspellings and tobi's point I approve this PR as there is no major point I think needs changing.

File: lua/terrortown/entities/items/item_ttt_armor.lua (hunk 8 above)
Comment (TimGoll) [reply lua/terrortown/entities/items/item_ttt_armor.lua:66]: Okay, Svens wording let me understand Tobis point. I'm pretty sure he meant the same thing. But there are two things: An armor system and an armor item.
I will add a new submenu called somthing like "Player Settings" where I add this stuff to. There are a handful more settings that could be moved there

File: lua/terrortown/lang/en.lua (hunk 10 above)
Comment (TimGoll) [reply lua/terrortown/lang/en.lua:1676]: You're right. Tht's what you get for copying this text from the TTT website :D

File: gamemodes/terrortown/entities/weapons/weapon_zm_improvised.lua (hunk 6 above)
Comment (TimGoll) [reply gamemodes/terrortown/entities/weapons/weapon_zm_improvised.lua]: done

File: lua/terrortown/menus/gamemode/administration/hud.lua (hunk 15 above)
Comment (TimGoll) [reply lua/terrortown/menus/gamemode/administration/hud.lua:81]: image

yes

File: lua/terrortown/entities/items/item_ttt_armor.lua (hunk 8 above)
Comment (TimGoll) [reply lua/terrortown/entities/items/item_ttt_armor.lua]: done.

**Review by saibotk — APPROVED** [review]
